scipy/sparse/csgraph/_flow.pyx

```diff
@@ -407,3 +426,207 @@ cdef ITYPE_t[:] _edmonds_karp(
             # If no augmenting path could be found, we're done.
             break
     return flow
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cdef bint _build_level_graph(
+        ITYPE_t[:] edge_ptr,
+        ITYPE_t source,
+        ITYPE_t sink,
+        ITYPE_t[:] capacities,
+        ITYPE_t[:] heads,
+        ITYPE_t[:] levels, 
```

Comment:
Qualifying arguments with `const` and sorting them accordingly with comments makes the use of data-structures more explicit.

```python
        const ITYPE_t[:] edge_ptr,      # IN
        const ITYPE_t source,           # IN
        const ITYPE_t sink,             # IN
        const ITYPE_t[:] capacities,    # IN
        const ITYPE_t[:] heads,         # IN
        ITYPE_t[:] levels,              # IN/OUT
```

The same suggestions apply to other interfaces.